src/systems/a320_systems/src/hydraulic.rs

```diff
@@ -1436,97 +1454,96 @@ impl A320AutobrakeController {
-            return false;
+        match self.mode {
+            AutobrakeMode::NONE => false,
+            AutobrakeMode::LOW | AutobrakeMode::MED => {
+                self.deceleration_demanded()
+                    && self
+                        .deceleration_governor
+                        .is_on_target(Ratio::new::<percent>(
+                            Self::MARGIN_PERCENT_TO_TARGET_TO_SHOW_DECEL_IN_LO_MED,
+                        ))
+            }
+            _ => {
+                self.deceleration_demanded()
+                    && self.deceleration_governor.decelerating_at_or_above_rate(
+                        Acceleration::new::<meter_per_second_squared>(
+                            Self::TARGET_TO_SHOW_DECEL_IN_MAX_MS2,
+                        ),
+                    )
+            }
         }
+    }
 
-        let deceleration_demanded = self.deceleration_governor_is_engaged()
-            && self.target.get::<meter_per_second_squared>() < 0.;
-
-        if self.mode == AutobrakeMode::LOW || self.mode == AutobrakeMode::MED {
-            deceleration_demanded
-                && self
-                    .deceleration_governor
-                    .is_on_target(Self::MARGIN_PERCENT_TO_TARGET_TO_SHOW_DECEL_IN_LO_MED)
-        } else {
-            deceleration_demanded
-                && self
-                    .deceleration_governor
-                    .deceleration_value()
-                    .get::<meter_per_second_squared>()
-                    < Self::TARGET_TO_SHOW_DECEL_IN_MAX_MS2
-        }
+    fn deceleration_demanded(&self) -> bool {
+        self.deceleration_governor.is_engaged()
+            && self.target.get::<meter_per_second_squared>() < 0.
     }
 
-    fn pedal_input_should_disarm(&self) -> bool {
+    fn should_disarm_due_to_pedal_input(&self) -> bool {
         match self.mode {
             AutobrakeMode::NONE => false,
             AutobrakeMode::LOW | AutobrakeMode::MED => {
                 // LOW and MED deactivates if one pedal over 53% or two over 11%.
-                self.left_brake_pedal_input > 0.53
-                    || self.right_brake_pedal_input > 0.53
-                    || (self.left_brake_pedal_input > 0.11 && self.right_brake_pedal_input > 0.11)
+                self.left_brake_pedal_input > Ratio::new::<percent>(53.)
+                    || self.right_brake_pedal_input > Ratio::new::<percent>(53.)
+                    || (self.left_brake_pedal_input > Ratio::new::<percent>(11.)
+                        && self.right_brake_pedal_input > Ratio::new::<percent>(11.))
             }
             AutobrakeMode::MAX => {
                 // MAX deactivates if one pedal over 77% or two over 53%.
-                self.left_brake_pedal_input > 0.77
-                    || self.right_brake_pedal_input > 0.77
-                    || (self.left_brake_pedal_input > 0.53 && self.right_brake_pedal_input > 0.53)
+                self.left_brake_pedal_input > Ratio::new::<percent>(77.)
```

Comment:
Great! Note that keeping the comment makes it more likely that the comment and actual code unalign at some point in the future. Therefore I'd consider removing the comments.